content/posts/cicd_ha.md

```diff
@@ -0,0 +1,132 @@
+
+As an example, in the image below we can see that a pipeline execution on a default Jenkins X deployment recovered just in 30 seconds to continue running after a [Pod](https://kubernetes.io/docs/concepts/workloads/pods/pod/) failure simulation. Nothing needed to be done in terms of deployment or pipeline definition. It is just its native behavior.
+
+![Jenkins X pod recovery](/img/cicd-ha/jenkins-x_podrestart.png)
+
+## The message and conclusion
+
+I've received in the past several redundant requests from organizations like the following ones:
+
+* *"We need to replicate and load balance our masters to have HA"*
+* *"My masters need to be up and running all the time as Active-Active to have a continuously running CI/CD service"*
+* *"How can I write to my shared storage at the same time with replicated masters?"* (another active-active HA conversation)
+
+And many more similar.
+
+The funny thing is that lots of cases were not prioritizing scalability, pipeline and master decoupling, or containerized orchestrated deployments. And then you realize that most of these requests come from IT requirements definitions in terms of HA general software architectures, no matter the nature of the service. 
+
+Most of the people asking about HA is always referring to an *Active-Active* replication scenario and usually thinking of infrastructure, not the real business issue when comes to CI/CD. 
+
+So it is very important to clarify that HA for CI/CD is not a matter of infrastructure replication. It is just a matter of resilient automated pipelines execution where usually just a couple of minutes of loss of service at some specific point is not even an issue... but you need to recover quickly and be stable on time.
+
+I cannot imagine a car manufacturer like BMW or Toyota replicating their entire manufacturing plants to support loss of service for HA. True that is not comparable in costs to run a couple more CPUs, but the same decoupling, flexibility and scalability best practices apply to get the same HA experience as CI/CD.
+
+Let's change our mindset. If we provide a scalable and distributed solution that deploys natively on a resilient platform, we don't even need to think about HA.
```

Comment:
This last line is a great way to summarize the post - I can see this being used as a highlighted quote for LinkedIn, Twitter, etc.

Reply:
Thanks @mdelgin for corrections. All looks good. Just done with all of them